server/src/main/java/org/elasticsearch/cluster/metadata/IndexTemplateMetaData.java

```diff
@@ -77,14 +79,13 @@
 
     private final Settings settings;
 
-    // the mapping source should always include the type as top level
     private final ImmutableOpenMap<String, CompressedXContent> mappings;
 
     private final ImmutableOpenMap<String, AliasMetaData> aliases;
 
     public IndexTemplateMetaData(String name, int order, Integer version,
                                  List<String> patterns, Settings settings,
-                                 ImmutableOpenMap<String, CompressedXContent> mappings,
+                                 CompressedXContent mappings,
```

Comment:
It is not very clear if we expect `CompressedXContent  mappings` to contain the type or not?  From line 98, it looks like it should not contain type, but then how `IndexTemplateMetaData` is built  `fromXContent` and ` builder` it looks like `mappings` already contain a type? 
Or we can expect both cases?